charts/datadog/templates/otel-agent-rbac.yaml

```diff
@@ -0,0 +1,40 @@
+{{- if and .Values.agents.rbac.create (eq (include "should-enable-otel-agent" .) "true") .Values.datadog.otelCollector.rbac.create -}}
```

Comment:
Note to reviewers: `should-enable-otel-agent` defined [here](https://github.com/DataDog/helm-charts/blob/c65f22450ed666129286c2123d0de864d3b26a90/charts/datadog/templates/_helpers.tpl#L112-L118)